sky/execution.py

```diff
@@ -284,6 +293,11 @@ def _execute(
             logger.info('Setup commands skipped.')
         elif Stage.SETUP in stages:
             backend.setup(handle, task, detach_setup=detach_setup)
+            backend.run_on_head(
+                handle,
+                _MAYBE_SKYLET_RESTART_CMD,
+                use_cached_head_ip=False,
+            )
```

Comment:
Keep this in `cloud_vm_ray_backend._provision`. We can have it at https://github.com/skypilot-org/skypilot/blob/b990a3a24fae2296e9f151efd2611b4058289854/sky/backends/cloud_vm_ray_backend.py#L2332